Source/Bogus/DataSets/Date.cs

```diff
@@ -144,31 +129,221 @@ public DateTimeOffset FutureOffset(int yearsToGoForward = 1, DateTimeOffset? ref
       /// <param name="end">End time</param>
       public DateTime Between(DateTime start, DateTime end)
       {
-         var minTicks = Math.Min(start.Ticks, end.Ticks);
-         var maxTicks = Math.Max(start.Ticks, end.Ticks);
+         ComputeRealRange(ref start, ref end, start.Kind, out var preferRangeBoundary);
+
+         var startTicks = start.ToUniversalTime().Ticks;
```

Comment:
I don't know all the ins and outs, but I think it is still [possible to hit problematic transitions using UTC](https://stackoverflow.com/a/40626617).
Could a HPET help with this?

Reply:
This is a good point, functions that call `SystemClock` multiple times should only call it once and use it as a consistent reference. Otherwise, in extremely rare cases, two consecutive calls will return different values, and the second one _could_ be earlier than the first one. I will make this change.